### PR TITLE
Fix highlighter support in PinnedQuery and added test

### LINKED ISSUE #53699: Highlight for Pinned Query not Working
**Elasticsearch version** (`bin/elasticsearch --version`): 7.4.2, 7.6.1

**Plugins installed**: repository-s3

**JVM version** (`java -version`):
`openjdk 13.0.2 2020-01-14
OpenJDK Runtime Environment AdoptOpenJDK (build 13.0.2+8)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 13.0.2+8, mixed mode, sharing)`

**OS version** (`uname -a` if on a Unix-like system):
Docker (from docker.elastic.co): `Linux ce1452e52a27 4.19.76-linuxkit #1 SMP Thu Oct 17 19:31:58 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`

**Description of the problem including expected versus actual behavior**:
Highlighting does not work when wrapping queries using the Pinned query, even though the issue at https://github.com/elastic/elasticsearch/issues/44074 describes pinned results as being highlight-able. At the very least, the organic results should be highlighted.
**Steps to reproduce**:

Please include a *minimal* but *complete* recreation of the problem, including
(e.g.) index creation, mappings, settings, query etc.  The easier you make for
us to reproduce it, the more likely that somebody will take the time to look at it.

 ```
POST _bulk
{ "index" : { "_index" : "test", "_id" : "1" } }
{ "title" : "elasticsearch is great" }
{ "index" : { "_index" : "test", "_id" : "2" } }
{ "title" : "kibana is pretty cool too" }
{ "index" : { "_index" : "test", "_id" : "3" } }
{ "title" : "another elasticsearch document" }

# Highlighting doesn't work with this:
GET test/_search
{
  "highlight": {
    "fields": {
      "title": {}
    }
  }, 
  "query": {
    "pinned": {
      "ids": ["3"],
      "organic": {
        "match": {
          "title": "elasticsearch"
        }
      }
    }
  }
}

# Highlighting works with the organic query:
GET test/_search
{
  "highlight": {
    "fields": {
      "title": {}
    }
  }, 
  "query": {
    "match": {
      "title": "elasticsearch"
    }
  }
}
```

### PR DESCRIPTION
**Pull request by markharwood**
CappedScoreQuery was missing visit(QueryVisitor visitor) delegation to wrapped query.
Closes #53699

### CODE REVIEW

File: x-pack/plugin/search-business-rules/src/main/java/org/apache/lucene/search/CappedScoreQuery.java [hunk 1]
```diff
@@ -33,6 +33,11 @@ public Query getQuery() {
         return query;
     }
 
+    @Override
+    public void visit(QueryVisitor visitor) {
+        query.visit(visitor);
```
Comment (jimczi) [inline x-pack/plugin/search-business-rules/src/main/java/org/apache/lucene/search/CappedScoreQuery.java]: @romseygeek should it be:
```suggestion
        visitor.getSubVisitor(BooleanClause.Occur.MUST, this))
```
?

**Review by jimczi — COMMENTED** [review]
I left one question

File: x-pack/plugin/search-business-rules/src/main/java/org/apache/lucene/search/CappedScoreQuery.java (hunk 1 above)
Comment (romseygeek) [reply x-pack/plugin/search-business-rules/src/main/java/org/apache/lucene/search/CappedScoreQuery.java]: Yes, that's a better option - this way a visitor can be informed of the existence of the CappedScoreQuery in the tree

**Review by jimczi — APPROVED** [review]
LGTM

**Pull request merged**